nixos/modules/installer/tools/nixos-option/nixos-option.cc

```diff
@@ -0,0 +1,588 @@
+//   ...
+//   users.users.avahi.uid = 10;
+//   users.users.avahi.useDefaultShell = false;
+//   users.users.cups.createHome = false;
+//   ...
+//   users.users.cups.useDefaultShell = false;
+//   users.users.gdm = ... ... ...
+//   users.users.messagebus = ... .. ...
+//   users.users.nixbld1 = ... .. ...
+//   ...
+//   users.users.systemd-timesync = ... .. ...
+void mapConfigValuesInOption(
+    const std::function<void(std::string const &path,
+                             std::variant<Value, Error> v)> &f,
+    std::string const &path, Context *ctx) {
+  Value *option;
+  try {
+    option =
+        findAlongAttrPath(*ctx->state, path, *ctx->autoArgs, ctx->config_root);
+  } catch (Error &e) {
+    f(path, e);
+    return;
+  }
+  recurse(
+      [f, ctx](std::string const &path, std::variant<Value, Error> v) {
+        bool leaf = std::holds_alternative<Error>(v) ||
+                    std::get<Value>(v).type != tAttrs ||
+                    ctx->state->isDerivation(std::get<Value>(v));
+        if (!leaf) {
+          return true; // Keep digging
+        }
+        f(path, v);
+        return false;
+      },
+      ctx, *option, path);
+}
+
+std::string describeError(Error const &e) { return "«error: " + e.msg() + "»"; }
+
+void describeDerivation(Context *ctx, Out &out, Value v) {
+  // Copy-pasted from nix/src/nix/repl.cc  :(
+  Bindings::iterator i = v.attrs->find(ctx->state->sDrvPath);
+  PathSet pathset;
+  try {
+    Path drvPath = i != v.attrs->end()
+                       ? ctx->state->coerceToPath(*i->pos, *i->value, pathset)
+                       : "???";
+    out << "«derivation " << drvPath << "»";
+  } catch (Error &e) {
+    out << describeError(e);
+  }
+}
+
+Value parseAndEval(EvalState *state, std::string const &expression,
+                   std::string const &path) {
+  Value v{};
+  state->eval(state->parseExprFromString(expression, absPath(path)), v);
+  return v;
+}
+
+void printValue(Context *ctx, Out &out, std::variant<Value, Error> maybe_value,
+                std::string const &path);
+
+void printUnsortedList(Context *ctx, Out &out, Value &v) {
+  Out list_out(out, "[", "]", v.listSize());
+  for (unsigned int n = 0; n < v.listSize(); ++n) {
+    printValue(ctx, list_out, *v.listElems()[n], "");
+    list_out << Out::sep;
+  }
+}
+
+void printSortedList(Context *ctx, Out &out, Value &v) {
+  std::vector<std::string> results;
+  for (unsigned int n = 0; n < v.listSize(); ++n) {
+    std::ostringstream buf;
+    Out buf_out(buf);
+    printValue(ctx, buf_out, *v.listElems()[n], "");
+    results.push_back(buf.str());
+  }
+  std::sort(results.begin(), results.end());
+  Out list_out(out, "[", "]", v.listSize());
+  for (auto const &v : results) {
+    list_out << v << Out::sep;
+  }
+}
+
+bool shouldSort(Context *ctx, Value &v) {
+  // Some lists should clearly be printed in sorted order, like
+  // environment.systemPackages.  Some clearly should not, like
```

Comment:
Hm, I don't think *any* lists should be sorted. E.g. the order of `environment.systemPackages` matters with respect to file collisions.